src/mam4xx/aero_model.hpp

```diff
@@ -0,0 +1,634 @@
+                          //! each rain droplet radius bin [1/s]
+
+  // outer loop for rain drop radius
+  for (int jr = 0; jr < nr; ++jr) {
+    // rain droplet radius
+    // rain droplet and aerosol particle radius [cm]
+    Real r_rain = rrainsv[jr];
+    // rain droplet fall speed [cm/s]
+    Real vfall = vfallrainsv[jr];
+    // inner loop for aerosol particle radius
+    Real scavsumnumbb = zero;
+    Real scavsumvolbb = zero;
+    for (int ja = 0; ja < na; ++ja) {
+      // aerosol particle radius
+      // rain droplet and aerosol particle radius [cm]
+      const Real r_aer = raerosv[ja];
+      Real etotal = zero; // efficiency of total scavenging effects [fraction]
+      calc_impact_efficiency(r_aer, r_rain, temp,       // & ! in
+                             freepath, rhoaero, rhoair, // & ! in
+                             vfall, airkinvisc,         // & ! in
+                             etotal);                   // out
+
+      // rain droplet sweep out volume [cm3/cm3/s]
+      const Real rainsweepout =
+          xnumrainsv[jr] * 4 * pi * r_rain * r_rain * vfall;
+      scavsumnumbb += rainsweepout * etotal * fnumaerosv[ja];
+      scavsumvolbb += rainsweepout * etotal * fvolaerosv[ja];
+    } // ja_loop
+
+    scavsumnum += scavsumnumbb;
+    scavsumvol += scavsumvolbb;
+
+  } // jr_loop
+
+  scavratenum = scavsumnum * 3600;
+  scavratevol = scavsumvol * 3600;
+
+} // end calc_1_impact_rate
+
+KOKKOS_INLINE_FUNCTION
+void modal_aero_bcscavcoef_init(
+    const Real dgnum_amode[AeroConfig::num_modes()],
+    const Real sigmag_amode[AeroConfig::num_modes()],
+    const Real aerosol_dry_density[AeroConfig::num_modes()],
+    Real scavimptblnum[nimptblgrow_total][AeroConfig::num_modes()],
+    Real scavimptblvol[nimptblgrow_total][AeroConfig::num_modes()]) {
+  // -----------------------------------------------------------------------
+  //
+  //  Purpose:
+  //  Computes lookup table for aerosol impaction/interception scavenging rates
+  //
+  //  Authors: R. Easter
+  //
+  // -----------------------------------------------------------------------
+  // @param [in]   dgnum_amode aerosol diameters [m]
+  // @param [in]   sigmag_amode standard deviation of aerosol size distribution
+  // @param [out]  scavimptblnum scavenging rate of aerosol number [1/s]
+  // @param [out]  scavimptblnum scavenging rate of aerosol volume [1/s]
+  // @aerosol_dry_density [in] aerosol dry density [k/m3]
+  // FIXME : create an 4 elements array that contains the aerosol densities to
+  // replace specdens_amode and lspectype_amode
+
+  const Real zero = 0;
+  const Real one = 1;
+  const Real three = 3;
+  // BAD CONSTANT
+  const Real dlndg_nimptblgrow = haero::log(1.25);
+  // ! set up temperature-pressure pair to compute impaction scavenging rates
+  const Real temp_0C = haero::Constants::melting_pt_h2o; //     ! K
+  // BAD CONSTANT
+  const Real press_750hPa = 0.75e6; //  ! dynes/cm2
```

Comment:
This seems to be storing the pressure p = 750 hPa expressed in dynes/cm2. In that sense it's just a reference pressure and not necessarily a `BAD CONSTANT`.